src/components/Header/Header.tsx

```diff
@@ -0,0 +1,115 @@
+import { Link } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import './Header.scss';
+import '../../styles/main.scss';
+
+export const Header: React.FC = () => {
+  const [isMenuVisible, setIsMenuVisible] = useState<boolean>(false);
```

Comment:
Why don't you create a custom hook for this logic? 
Like, `useOpenMenu` or something, it's definitely not required, but may be a good practice, just for you to get that feeling of creating custom hooks for logic encapsulation.